base/math.jl

```diff
@@ -322,6 +322,10 @@ end
 # until the heuristics can be improved
 @inline nan_dom_err(out, x) = isnan(out) & !isnan(x) ? throw(DomainError(x, "NaN result for non-NaN input.")) : out
 
+# utility for reporting error when a function is evaluated outside its domain TOOD: upper bound and double bound
+@noinline throw_dom_err(f, x, low) = throw(DomainError(x, "$f(x) called with x < $low"))
```

Comment:
Maybe something like "$f($x) is invalid as $f does not accept x < $low" or "$x is not valid for $f(x) (x must be > $low)" or a variant along these lines? I think it's really nice to say that this is invalid (even if it's kind of redundant, it helps understanding the message), and to print the value of `x` (which is sometimes not what the caller imagined).

Reply:
The value of `x` is printed since `x` is passed to the DomainError constructor.

Reply:
Ah, good. I still think it would be nice to state that "x < $low" is invalid. Messages that don't say what's wrong always look weird to me.